zarp/run/snakemake.py

```diff
@@ -1 +1,126 @@
 """Module for executing Snakemake workflows."""
+
+import subprocess
+from typing import List, Optional
+
+from zarp.config.models import (ExecModes, InitRun,
+                                DependencyEmbeddingStrategies)
+
+
+class SnakemakeExecutor:
+    """Run snakemake with system calls.
+
+    Args:
+        run (InitRun): Run-specific parameters.
+
+    Attributes:
+        run_dict (dict): Dictionary with run-specific parameters.
+        success (bool): Indicator for successful snakemake run.
+        run_list (list[str]): List containing strings for snakemake call.
+
+    Example:
+        The example below expects a valid `Snakefile` (e.g. `touch Snakefile`)
+        in the current working directory.
+        It constructs a run with default values and runs it.
+        >>> mysnk = SnakemakeExecutor(InitRun())
+        >>> mysnk.prepare_run(snkfile = "Snakefile")
+        >>> mysnk.run()
+        >>> assert mysnk.get_success()
+
+    """
+
+    def __init__(self, run: InitRun) -> None:
+        """Class constructor."""
+        self.run_dict = run.dict()
+        self.success: Optional[bool] = None
```

Comment:
`Optional[bool]` seems weird to me. I will add an enumerator with states instead, as it is more extensible. For now, the states will be `UNKNOWN`, `SUCCESS` and `ERROR`.